mam4_refactor_scripts/beg.inp

```diff
@@ -1,69 +1,92 @@
   ! index (y_k or klev) to be passed to this routine and
   ! uncomment the following code:
   ! integer, intent(in) :: y_i, y_k, y_lchnk
+  !-----------------------------------------------------------------------------------------
 
-  !>
+  !-----------------------------------------------------------------------------------------
+  ! This is used when multiple sets of yaml output is needed
+  !to cover different options (e.g., true and false)
+  ! character(len=200) :: ext_str
+  !-----------------------------------------------------------------------------------------
 
-  ! character(len=200) :: ext_str  ! this is used when multiple sets of yaml output is needed to cover different options (e.g., true and false)
   type(yaml_vars) :: yaml
   integer  :: unit_input, unit_output, y_nstep
-  integer,save :: n_calls=0   ! some subroutines are called multiple times in one timestep, record the number of calls
+
+  ! some subroutines are called multiple times in one timestep, record the number of calls
+  integer,save :: n_calls=0
+
 
   !populate YAML structure
-  yaml%lev_print = <Add hardwired level here> !level (**remove these if generating data for a dependent subroutines**)
-  yaml%nstep_print = <add hardwired time step here> !time step(**remove these if generating data for a dependent subroutines**)
+  !(**remove yaml%lev_print, nstep_print, col_print if generating data for a dependent subroutines**)
+  yaml%lev_print = <Add hardwired level here>       !level
+  yaml%nstep_print = <add hardwired time step here> !time step
 
-  ! set ext_str if there are multiple sets of yaml output to be write out
-  ! here gives an example that "flag" in the code can be 0, 1, or 2:
-  ! write(ext_str,'(I2)') flag
-  ! ext_str = 'flag_'//adjustl(ext_str)
+  yaml%col_print = icolprnt(y_lchnk)                !column to write data
 
+  !current_time step
+  y_nstep = get_nstep()
 
-  !YAML file input generation code- DO NOT PORT to C++
-  !print all inputs one-by-one at column "yaml%col_print"
-  yaml%col_print = icolprnt(y_lchnk) !column to write data(**remove these if generating data for a dependent subroutines**)
-  y_nstep = get_nstep() !time step (**remove these if generating data for a dependent subroutines**)
+  !Flag to decide to write or not to write data
+  yaml%flag_print = .false. !(**remove these if generating data for a dependent subroutines**)
 
-  yaml%flag_print = .false. ! to write or not to write data (**remove these if generating data for a dependent subroutines**)
   if(yaml%col_print == y_i .and. y_nstep==yaml%nstep_print .and. y_k == yaml%lev_print) then ! if this column exists in y_lchnk
-     !<
-     !In the case of y_i or y_k are not passed as arguments:
+
+     !-----------------------------------------------------------------------------------------
+     !In the case of y_i or y_k are not passed as arguments, use the following if condition:
      !if(yaml%col_print >0 .and. y_nstep==yaml%nstep_print) then
```

Comment:
maybe reduce the indent to the previous line so we know that they are alternate options of the above if condition, other than an inner loop.